packages/bpk-component-datatable/src/hasChildrenOfType.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Children } from 'react';
+
+export function getDisplayName(Component) {
+  return (
+    Component.displayName ||
+    Component.name ||
+    (typeof Component === 'string' && Component.length > 0
+      ? Component
+      : 'Unknown')
+  );
+}
+
+const hasChildrenOfType = (type, atLeast = 1) => (props, propType, componentName) => {
+  if (Children.count(props[propType]) < atLeast) {
+    const inflectedNoun = atLeast === 1 ? 'child' : 'children';
+    return Error(
+      `${componentName} requires at least ${atLeast} '${getDisplayName(type)}' ${inflectedNoun}.`,
+    );
+  }
+
+  const children = Children.toArray(props[propType]);
+  for (let i = 0; i < children.length; i += 1) {
```

Comment:
You can use `React.Children.forEach` here: https://reactjs.org/docs/react-api.html#reactchildrenforeach

Reply:
The reason for this approach is to be able to return directly out of the iteration block -- can you do `return React.Children.forEach(...)`?

Reply:
Ah yeah, good point. I'm surprised eslint didn't object to that, it usually shouts at me when I use a native `for` loop. But as long as it works for you 😄

Reply:
Yeah i was waiting to hear from it :)